File: interviewbit-strings-integer-to-roman.py
class Solution:
    # @param A : integer
    # @return a strings
    def intToRoman(self, A):
        codebook = [[1000, "M"], [900, "CM"], [500, "D"], [400, "CD"], [100, "C"], [90, "XC"], [50, "L"], [40, "XL"],
                    [10, "X"], [9, "IX"], [5, "V"], [4, "IV"], [1, "I"]]
        roman_numeral = ""
        for decimal, roman in codebook:
            while A >= decimal:
                roman_numeral += roman
                A -= decimal

        return roman_numeral

File: interviewbit-strings-integer-to-roman.java
public class Solution {
    public String intToRoman(int a) {
        int[] decimals = {1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1};
        String[] romans = {"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"};
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < decimals.length; i++) {
            while (a >= decimals[i]) {
                a -= decimals[i];
                sb.append(romans[i]);
            }
        }
        return sb.toString();
    }
}
